ext/mysqli/mysqli_warning.c

```diff
@@ -199,49 +199,50 @@ PHP_METHOD(mysqli_warning, next)
 /* }}} */
 
 /* {{{ property mysqli_warning_message */
-static
-zval *mysqli_warning_message(mysqli_object *obj, zval *retval)
+static int mysqli_warning_message(mysqli_object *obj, zval *retval, zend_bool quiet)
 {
 	MYSQLI_WARNING *w;
 
 	if (!obj->ptr || !((MYSQLI_RESOURCE *)(obj->ptr))->ptr) {
-		return NULL;
+		return FAILURE;
```

Comment:
I think we should either be throwing a warning here, or doing ZVAL_NULL(retval) and returning SUCCESS. Having a failure here without warning is odd.

Reply:
I went with the latter as I didn't want to introduce a new warning in a patch release. But I am also ok with a warning if that's better.

However, thinking out loud... If a warning is introduced now, then it can also be reclassified to an exception for PHP 8 which would be more in line with other property access errors.

Reply:
I'd try the warning and see if the test fallout is bad :) Being able to make that an exception (as this seems to be in all other cases) would be nice.